src/Core/Generator/SchemaGeneratorFactory.cs

```diff
@@ -56,46 +58,78 @@ public static async Task<string> Create(RuntimeConfig config, string mode, int?
+
+            config.DataSource?.Options?.TryGetValue("database", out globalDatabaseName);
+            config.DataSource?.Options?.TryGetValue("container", out globalContainerName);
+
+            if (string.IsNullOrEmpty(connectionString) || globalDatabaseName is null)
             {
-                logger.LogError("Connection String, Database and container name must be provided in the config file");
-                throw new ArgumentException("Connection String, Database and container name must be provided in the config file");
+                logger.LogError("Connection String and Database name must be provided in the config file");
+                throw new ArgumentException("Connection String and Database name must be provided in the config file");
             }
 
-            logger.LogInformation("Connecting to Cosmos DB Database: {0}, Container: {1}", databaseName, containerName);
+            HashSet<string> dbAndContainerToProcess = new();
+            if (globalDatabaseName != null && globalContainerName != null)
+            {
+                dbAndContainerToProcess.Add($"{globalDatabaseName}.{globalContainerName}");
+            }
 
-            // Connect to the Azure Cosmos DB container.
-            Container container = ConnectToCosmosDB(connectionString, databaseName, containerName);
-            SamplingModes samplingMode = (SamplingModes)Enum.Parse(typeof(SamplingModes), mode);
+            foreach (KeyValuePair<string, Entity> entity in config.Entities)
+            {
+                dbAndContainerToProcess.Add(entity.Value.Source.Object);
+            }
 
-            // Determine the appropriate sampler based on the sampling mode.
-            ISchemaGeneratorSampler schemaGeneratorSampler = samplingMode switch
+            StringBuilder schema = new();
+            foreach (string dbAndContainer in dbAndContainerToProcess)
             {
-                SamplingModes.TopNExtractor => new TopNExtractor(container, sampleCount, days, logger),
-                SamplingModes.EligibleDataSampler => new EligibleDataSampler(container, partitionKeyPath, sampleCount, days, logger),
-                SamplingModes.TimePartitionedSampler => new TimePartitionedSampler(container, groupCount, sampleCount, days, logger),
-                _ => throw new ArgumentException($"Invalid sampling mode: {mode}, Valid Sampling Modes are: {SamplingModes.TopNExtractor}, {SamplingModes.EligibleDataSampler}, {SamplingModes.TimePartitionedSampler}")
-            };
+                if (string.IsNullOrEmpty(dbAndContainer)) // Safety check
+                {
+                    continue;
+                }
 
-            logger.LogInformation("Sampling Started using {0}", schemaGeneratorSampler.GetType().Name);
+                string[] dbContainer = dbAndContainer.Split('.');
+                if (dbContainer.Length != 2)
+                {
+                    logger.LogError("Invalid Database and Container Name provided in the config file");
```

Comment:
this is mandating entity source name to always be in the format of database.container. What if I dont want to provide database name, and would like to rely on globaldatabase name? In that situation, I should be able to simply use the source name of entity as the container name and rely on global databasename.

Reply:
fixed.